### ISSUE TITLE
MultiLogloss: support NaN in target matrix for multi-task learning with incomplete labels

### ISSUE THREAD

**Issue by ismaelropa** [catboost/catboost#3079]
## Description
In multi-task / multi-endpoint classification, secondary task labels are often
only partially available — a sample may be labeled for the primary task but not
for task k. The current implementation rejects any NaN in the MultiLogloss label
matrix with:

  catboost/private/libs/target/target_converter.cpp:629:
  Target Labels for MultiLogloss must be 0 or 1

## Requested behaviour
NaN entries in the label matrix should be silently masked (that sample–task pair
contributes 0 gradient for task k). This is how XGBoost handles missing targets
in multi-output objectives (`missing=NaN`).

## CatBoost version
1.2.10

## Minimal reproducer
```python
import catboost as cb
import numpy as np

X = np.random.rand(10, 5)
y = np.column_stack([
    np.random.randint(0, 2, 10).astype(float),  # primary
    [0, 1, np.nan, 0, 1, np.nan, 0, 1, 0, 1],   # secondary — 2 NaN
])

model = cb.CatBoostClassifier(loss_function='MultiLogloss', verbose=False)
model.fit(X, y)  # raises: Target Labels for MultiLogloss must be 0 or 1
